### PR TITLE
Feature/base token tests 164

### PR DESCRIPTION
**Pull request by AnnRiera**
Base token tests implementation.

### CODE REVIEW

File: src/base-token.spec.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      stub.withArgs(AddressZero, 1).onCall(0).rejects()
+
+      expect(provider._getBalance(AddressZero, 1)).to.be.rejectedWith(Error)
+      stub.restore()
+    })
+  })
+
+  describe('setApprovalForAll', () => {
+    it('should grants or revokes permission to addressOperator to transfer the callers tokens, according to approved', async () => {
+      const stub = sinon.stub(BaseToken.prototype, 'setApprovalForAll')
```
Comment (bzuker) [inline src/base-token.spec.ts:58]: similar to the comment above, I don't think we should be stubbing the method itself. In these cases, what I think we should check is that `this.baseContract.methods.setApprovalForAll` is called with the correct params and nothing else.

**Review by bzuker — COMMENTED** [review]
Left a few comments, regarding how I think we should test these methods. We can discuss offline if I wasn't clear enough, let me know!

File: src/base-token.spec.ts (hunk 1 above)
Comment (AnnRiera) [reply src/base-token.spec.ts]: Copy that!

**Review by jarcodallo — APPROVED** [review]
LGTM

**Pull request merged**